query/storage.go

```diff
@@ -24,6 +24,17 @@ type StorageReader interface {
 	Close()
 }
 
+type GroupCapability interface {
+	HaveCount() bool
+	HaveSum() bool
+	HaveFirst() bool
+	HaveLast() bool
+}
+
+type GroupAggregator interface {
```

Comment:
It seems like the corresponding interface for windowed aggregation is `WindowAggregateReader`. Should this be called something like `AggregatingGroupReader` or something like that?

Reply:
The reason I broke from convention here is that this interface doesn't have a "Read*" method. `ReadGroup` is part of the `StorageReader` interface, whereas `ReadAggregateWindow` is part of the `WindowAggregateReader` interface. I thought that might lead to some confusion, but I'm personally fine with either.